src/stan/variational/families/normal_fullrank.hpp

```diff
@@ -379,6 +379,26 @@ namespace stan {
         eta = transform(eta);
       }
 
+      /**
+        * Draw a posterior sample from the normal distribution, 
+        * and return its log normal density. The constant (d log 2 pi) is removed. 
+        * @param[in] rng Base random number generator.
+        * @param[in, out] eta Vector to which the draw is assigned; must
+        * already be properly sized. eta will be transformed into variational posteriors. 
+        * @param[out] log_q The log density in the full-rank normal approximation.
+        * @throws std::range_error If the index is out of range.
+        */
+      template <class BaseRNG>
+      void sample_log_q(BaseRNG& rng,
```

Comment:
Required: add unit tests for `sample_log_q()` for both this class and `normal_meanfield.hpp`. I'll write more as a general comment.

Reply:
unit test added for calc_log_g